server/src/main/java/org/opensearch/index/fielddata/fieldcomparator/BytesRefFieldComparatorSource.java

```diff
@@ -134,9 +134,11 @@ protected SortedDocValues getSortedDocValues(LeafReaderContext context, String f
         }
 
         return new FieldComparator.TermValComparator(numHits, null, sortMissingLast) {
+            LeafReaderContext leafReaderContext;
 
             @Override
             protected BinaryDocValues getBinaryDocValues(LeafReaderContext context, String field) throws IOException {
+                leafReaderContext = context;
```

Comment:
This is the only place that concerns: the `getBinaryDocValues` could be called from multiple index searcher / search threads, right?

Reply:
the `getBinaryDocValues` is called by the `getLeafComparator` which is called during the initialization of the `TopFieldLeafCollector` (instance of this is created per leaf) - we should be fine here.